cloud/pkg/synccontroller/synccontroller.go

```diff
@@ -223,6 +244,59 @@ func (sctl *SyncController) manageObjectSync(syncs []*v1alpha1.ObjectSync) {
 	}
 }
 
+// 	when CloudCore starts, it will delete outdated syncs, then triggered by node delete events.
+func (sctl *SyncController) DeleteOutdatedObjectSync() {
+	sctl.deleteObjectSyncs()
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop synccontroller DeleteOutdatedObjectSync loop")
+			return
+		case e := <-sctl.nodeManager.Events():
+			node, ok := e.Object.(*v1.Node)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", node)
+				continue
+			}
+			switch e.Type {
+			case watch.Deleted:
+				sctl.deleteObjectSyncs()
+			}
+		}
+	}
+}
+
+func (sctl *SyncController) deleteObjectSyncs() {
+	syncs, err := sctl.objectSyncLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("Failed to list all the ObjectSyncs: %v", err)
+	}
+	for _, sync := range syncs {
+		nodeName := getNodeName(sync.Name)
+		if isGarbage, err := sctl.checkObjectSync(nodeName); isGarbage {
+			klog.Infof("ObjectSync %v will be deleted because node %v unregisters  ", sync.Name, nodeName)
+			err = sctl.crdClient.ReliablesyncsV1alpha1().ObjectSyncs(sync.Namespace).Delete(sync.Name, metav1.NewDeleteOptions(0))
+			if err != nil {
+				klog.Fatalf("Fail to delete objectSync %v of EdgeNode %v which is offline", sync.Name, nodeName)
+			}
+			if err != nil {
+				klog.Fatalf("objectSync GC failed, err:%v", err)
+			}
+		}
+	}
+}
+
+// checkObjectSync checks whether objectSync is outdated
+func (sctl *SyncController) checkObjectSync(nodeName string) (isGarbage bool, err error) {
```

Comment:
Is there necessary to specify val name when return:
`func (sctl *SyncController) checkObjectSync(nodeName string) (bool, error) {`

Reply:
Seems the return val err never used, why not remove it?

Reply:
@daixiang0  yes,(bool, error) is better.  the return val err should be used, I will add some lines to use it.